src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,160 @@
+
+        private BoundBlock LowerBoundMultipleLocalDeclarationUsingVar(BoundMultipleLocalDeclarations boundMultiple,
+                                                                      ImmutableArray<LocalSymbol> locals,
+                                                                      List<BoundStatement> precedingStatements,
+                                                                      ImmutableArray<BoundStatement> followingStatements)
+        {
+            List<BoundLocalDeclaration> reversedLocals = Enumerable.Reverse(boundMultiple.LocalDeclarations).ToList();
+            List<BoundUsingStatement> reversedUsingStatements = new List<BoundUsingStatement>();
+            for (int i = 0; i < reversedLocals.Count; i++)
+            {
+                BoundBlock innerBlock; 
+                // The first element in the reversed lists' using statement must contain all following statements.
+                if (i == 0)
+                {
+                    innerBlock = new BoundBlock(
+                            syntax: boundMultiple.Syntax,
+                            locals: ImmutableArray.Create<LocalSymbol>(),
+                            statements: followingStatements
+                            );
+                }
+                // All other elements will only contain the previous element as a following statement.
+                else
+                {
+                    BoundStatement previousUsing = reversedUsingStatements.Last();
+                    innerBlock = new BoundBlock(
+                            syntax: boundMultiple.Syntax,
+                            locals: ImmutableArray.Create<LocalSymbol>(),
+                            statements: ImmutableArray.Create(previousUsing)
+                            );
+                }
+
+                BoundUsingStatement boundUsing = new BoundUsingStatement(
+                        syntax: boundMultiple.Syntax,
+                        locals: ImmutableArray.Create<LocalSymbol>(),
+                        declarationsOpt: new BoundMultipleLocalDeclarations(
+                            boundMultiple.Syntax,
+                            ImmutableArray.Create(boundMultiple.LocalDeclarations[i])),
+                        expressionOpt: null,
+                        iDisposableConversion: Conversion.Identity,
+                        disposeMethodOpt: null,
+                        body: innerBlock
+                        );
+                reversedUsingStatements.Add(boundUsing);
+            }
+
+            precedingStatements.Add((BoundStatement)reversedUsingStatements.Last());
+            BoundBlock outermostBlock = new BoundBlock(
+                            syntax: boundMultiple.Syntax,
+                            locals: locals,
+                            statements: precedingStatements.ToImmutableArray());
+            return outermostBlock;
+        }
+
+        internal static bool ContainsUsingVariable(BoundStatement boundStatement)
```

Comment:
The only use of this method I can find is in binding the switch section. If so, I think we should make this a private method in binding instead of the local rewriter.